yarn-project/aztec.js/src/utils/cheat_codes.ts

```diff
@@ -271,6 +286,96 @@ export class EthCheatCodes {
   }
 }
 
+/** Cheat codes for the L1 rollup contract. */
+export class RollupCheatCodes {
+  private client: WalletClient & PublicClient;
+  private rollup: GetContractReturnType<typeof RollupAbi, WalletClient>;
+
+  private logger = createDebugLogger('aztec:js:cheat_codes');
+
+  constructor(private ethCheatCodes: EthCheatCodes, private addresses: Pick<L1ContractAddresses, 'rollupAddress'>) {
+    this.client = createWalletClient({ chain: foundry, transport: http(ethCheatCodes.rpcUrl) }).extend(publicActions);
+    this.rollup = getContract({
+      abi: RollupAbi,
+      address: addresses.rollupAddress.toString(),
+      client: this.client,
+    });
+  }
+
+  /** Returns the current slot */
+  public async getSlot() {
+    const ts = BigInt((await this.client.getBlock()).timestamp);
+    return await this.rollup.read.getSlotAt([ts]);
+  }
+
+  /** Returns the current epoch */
+  public async getEpoch() {
+    const slotNumber = await this.getSlot();
+    return await this.rollup.read.getEpochAtSlot([slotNumber]);
+  }
+
+  /** Warps time in L1 until the next epoch */
+  public async advanceToNextEpoch() {
+    const slot = await this.getSlot();
+    const slotsUntilNextEpoch = BigInt(AZTEC_EPOCH_DURATION) - (slot % BigInt(AZTEC_EPOCH_DURATION)) + 1n;
+    const timeToNextEpoch = slotsUntilNextEpoch * BigInt(AZTEC_SLOT_DURATION);
+    const l1Timestamp = BigInt((await this.client.getBlock()).timestamp);
+    await this.ethCheatCodes.warp(Number(l1Timestamp + timeToNextEpoch));
+    this.logger.verbose(`Advanced to next epoch`);
+  }
+
+  /** Returns the current proof claim (if any) */
+  public async getProofClaim(): Promise<EpochProofClaim | undefined> {
+    // REFACTOR: This code is duplicated from l1-publisher
+    const [epochToProve, basisPointFee, bondAmount, bondProviderHex, proposerClaimantHex] =
+      await this.rollup.read.proofClaim();
+
+    const bondProvider = EthAddress.fromString(bondProviderHex);
+    const proposerClaimant = EthAddress.fromString(proposerClaimantHex);
+
+    if (bondProvider.isZero() && proposerClaimant.isZero() && epochToProve === 0n) {
+      return undefined;
+    }
+
+    return {
+      epochToProve,
+      basisPointFee,
+      bondAmount,
+      bondProvider,
+      proposerClaimant,
+    };
+  }
+
+  /**
+   * Marks the specified block (or latest if none) as proven
+   * @param maybeBlockNumber - The block number to mark as proven (defaults to latest pending)
+   */
+  public async markAsProven(maybeBlockNumber?: number | bigint) {
```

Comment:
I'm hoping we can swap the implementation of this to just write the storage slot of the proven tip and get rid of the assumeProven flow entirely.

Reply:
We'd also need to manually write the storage slots for the last `proofClaim`, distribute fees, and tie this into the AnvilWatcher if we want every new block (or epoch) to be auto-proven. But yeah, I'd rather leave the extra complexity out of the solidity contract.